cmd/utils/flags.go

```diff
@@ -758,13 +763,21 @@ func SetupNetwork(ctx *cli.Context) {
 		core.BlockReward = big.NewInt(1.5e+18)
 		core.ExpDiffPeriod = big.NewInt(math.MaxInt64)
 	}
+	params.TargetGasLimit = common.String2Big(ctx.GlobalString(TargetGasLimitFlag.Name))
 }
 
-// SetupVM configured the VM package's global settings
-func SetupVM(ctx *cli.Context) {
-	vm.EnableJit = ctx.GlobalBool(VMEnableJitFlag.Name)
-	vm.ForceJit = ctx.GlobalBool(VMForceJitFlag.Name)
-	vm.SetJITCacheSize(ctx.GlobalInt(VMJitCacheFlag.Name))
+// MakeChainConfig reads the chain configuration from the given database or returns
+// an error if it failed.
+func MakeChainConfig(db ethdb.Database, ctx *cli.Context) *core.ChainConfig {
```

Comment:
Coolio, you dropped the error. But could we also call it `MustMakeChainConfig` to emphasize that it can terminate the program (similar to the stdlib's [`MustCompile`](https://golang.org/pkg/regexp/#MustCompile) for example)?